abdiff/core/run_ab_transforms.py

```diff
@@ -1 +1,203 @@
+        if root.endswith("transformed"):
+            # skip 'transformed' root directory
+            continue
+        for filename in files:
+            relative_dir = root.replace(run_directory, "")
+            sub_filepaths.append(os.path.join(relative_dir, filename))
+        filepaths.append(sub_filepaths)
+    return tuple(filepaths)
+
+
+def get_logs(run_id: str, container_ids: list[str], log_file: str) -> str:
+    """Monitor Docker container runs and write logs to file as containers exit."""
+    logged_container_ids: set[str] = set()
+    logs = ""
+    while logged_container_ids != set(container_ids):
+        if not DockerClient.containers.list(all=True, filters={"label": f"run={run_id}"}):
+            raise DockerContainersNotFoundError(run_id)
+
+        # get logs from completed container runs for a given "run"
+        exited_containers = [
+            container
+            for container in DockerClient.containers.list(
+                filters={"label": f"run={run_id}", "status": "exited"}
+            )
+            if container.id in container_ids and container.id not in logged_container_ids
+        ]
+        if any(exited_containers):
+            for container in exited_containers:
+                logs += str(
+                    write_container_logs(
+                        container,
+                        log_file,
+                    )
+                )
+                logger.info(logs)
+                logged_container_ids.add(container.id)
+        time.sleep(5)
+    return logs
+
+
+def write_container_logs(container: Container, log_file: str) -> str:
+    """Get logs from container and write to file."""
+    labels = container.labels
+    logger.info(
+        f"Container '{container.id}' (Docker image: {labels["docker_image"]}) "
+        f"COMPLETED transform for '{labels["source"]}' "
+        f"input_file: {labels["input_file"]}."
+    )
+
+    # get logs
+    logs = container.logs()
+
+    # create section header for log file
+    header = (
+        f"docker_image: {labels['docker_image']} | "
+        f"source: {labels['source']} | "
+        f"input_file: {labels['input_file']}\n"
+    ).encode()
+
+    # write logs
+    with open(Path(log_file), "ab") as file:
+        file.write(header)
+        file.write(logs)
+        file.write(b"\n")
+    return logs
+
+
+def generate_transform_command(source: str, input_file: str) -> str:
+    """Generate transform command arguments for Transmogrifier.
+
+    This method will parse command arguments from the provided 'input_file'.
+    The generated transform command arguments are used when running container(s)
+    with the transmogrifier Docker images.
+    """
+    input_file_name = input_file.split("/")[-1]
+    input_file_ext = input_file_name.split(".")[-1]
+    output_file_name = input_file.split("/")[-1].replace(
+        f"extracted-records-to-index.{input_file_ext}",
+        "transformed-records-to-index.json",
+    )
```

Comment:
If we include the `source` [parsing above](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/6/files#diff-7a49d9dbf619fca67eb5ed5ed16b6c62440a4cc6898754cd835d2b819be74ddcR52), this is a lot of parsing and constructing based on the file naming convention for TIMDEX files.  

While I like the responsibility of this function to generate commands, perhaps it makes sense to break-out the filename parsing to a standalone function that could be tested?

I recently faced this while working on the parquet exploration repo, using [regex to parse the pieces of the filename](https://github.com/MITLibraries/dataeng-workspace-parquet/blob/main/pqsandbox/timdex/s3.py#L19-L36).  

Not sure it's totally applicable here... as you may not need to know the date or the cadence (as examples), but thought worth sharing.  This regex also handles the edge cases where an source like `alma` adds a `_01`, `_02`, etc., suffix to the filename before the file extension.  Though with what you have, think it would be fine (it'd just get copied).

So maybe... just a function like `get_transform_filename_from_extract_filename`?

Reply:
But maybe over-complicating things!  Feels as though this would work.

Reply:
Please see the added function [`parse_transform_details_from_extract_filename`](https://github.com/MITLibraries/transmogrifier-ab-diff/pull/6/files#diff-7a49d9dbf619fca67eb5ed5ed16b6c62440a4cc6898754cd835d2b819be74ddcR187-R202). Thank you for sharing the parsing method with regex! I ultimately went with the regex approach as it felt more robust but simplified the regex pattern for this use case.